.pre-commit-hooks.yaml

```diff
@@ -6,30 +6,30 @@
 - id: clang-format
   name: clang-format
   entry: clang-format-hook
-  description: Formats C/CPP code
-  files: \.(c|cc|cxx|cpp|h|hpp|hxx|m)$
+  description: Formats C, C++, Objective-C and Java code
+  files: \.(c|cc|cxx|cpp|h|hpp|hxx|m|mm|java)$
   language: python
 - id: clang-tidy
   name: clang-tidy
   entry: clang-tidy-hook
-  description: Find warnings/errors in C/CPP code
+  description: Find warnings/errors in C, C++ and Objective-C code
   files: \.(c|cc|cxx|cpp|h|hpp|hxx|m)$
   language: python
 - id: oclint
   name: oclint
   entry: oclint-hook
-  description: Find warnings/errors in C/CPP code
+  description: Find warnings/errors in C, C++ and Objective-C code
   files: \.(c|cc|cxx|cpp|h|hpp|hxx|m)$
   language: python
 - id: uncrustify
   name: uncrustify
   entry: uncrustify-hook
-  description: Formats C/CPP code
-  files: \.(c|cc|cxx|cpp|h|hpp|hxx|m)$
+  description: Formats C, C++, Objective-C, Java, D and Vala code
```

Comment:
Per uncrustify, this is the list `C, C++, C#, ObjectiveC, D, Java, Pawn, and VALA`